tinynn/util/bn_rebuild.py

```diff
@@ -0,0 +1,76 @@
+import typing
+
+import torch
+import torch.nn as nn
+import copy
+
+
+class ConvBnForward(nn.Module):
+    def __init__(self, conv):
+        super().__init__()
+
+        assert isinstance(conv, nn.Conv2d), "not a conv2d"
+        self.conv = conv
+        self.bn = nn.BatchNorm2d(conv.out_channels)
+
+    def forward(self, input_0):
+        # We forward bn in train mode, but do not use its output to get running stat in train_set.
+        conv = self.conv(input_0)
+        self.bn(conv)
+        return conv
+
+
+class ConvBnTrain(nn.Module):
+    def __init__(self, convbn):
+        super().__init__()
+        assert isinstance(convbn, ConvBnForward), "not a ConvBnForward"
+        self.conv = convbn.conv
+        self.bn = convbn.bn
+        eps = self.bn.eps
+        # Use the forward running_stat to get weight and bias of rebuilt bn.
+        weight = (self.bn.running_var + eps) ** 0.5
+        bias = self.bn.running_mean
+        self.bn.weight.data.copy_(weight)
+        self.bn.bias.data.copy_(bias)
+
+    def forward(self, input_0):
+        conv = self.conv(input_0)
+        bn = self.bn(conv)
+        return bn
+
+
+def add_bn(origin_model, layer_fused_bn):
+    model = copy.deepcopy(origin_model)
+    for name, mod in model.named_children():
+        if isinstance(mod, nn.Conv2d) and name in layer_fused_bn:
```

Comment:
You will need to do recursion in the `else` branch. Otherwise, the modules in a `nn.Sequential` object won't be updated.